Algorithms.Tests/Search/InterpolationSearchTest.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Linq;
+using Algorithms.Search;
+using NUnit.Framework;
+using NUnit.Framework.Internal;
+
+namespace Algorithms.Tests.Search;
+
+public class InterpolationSearchTest
+{
+    [Test]
+    public void Interpolation_Find_Index_Test([Random(1, 1000, 100)] int n)
+    {
+        // Arrange
+        var search = new InterpolationSearch();
+        var random = Randomizer.CreateRandomizer();
+        var arrayToSearch = Enumerable.Range(0, n).Select(_ => random.Next(0, 1000)).OrderBy(x => x).Distinct().ToArray();
```

Comment:
Why do you only select distinct values? The search should work with duplicates as well